publicsuffix/rules.go

```diff
@@ -8626,5 +8626,7 @@ func init() {
 		{1, "now.sh", 2, true},
 		{1, "zone.id", 2, true},
 	}
-	DefaultList.rules = r[:]
+	for i := range r {
+		DefaultList.AddRule(&r[i])
```

Comment:
What's the reason of this change? Is it because of #141?

Reply:
OK, nevermind. I actually figured it out, we need to pass via `AddRule` because the internal `rule` is no longer a list.

My bad, sorry for the stupid question.

I need to determine the cost of this operation. It could be possible that it's more efficient to compile directly the `r` in the appropriate form and then assign.

Reply:
I can't speak with absolute certainty, but my gut feeling is that this implementation eases the load on the garbage collector because nothing in this list should ever be garbage collected which means it's better to have all of them in a single giant array rather than allocating memory for each individually on the heap. I might be wrong and it might be worth benchmarking.